android-components/components/support/ktx/src/test/java/mozilla/components/support/ktx/android/view/ActivityTest.kt

```diff
@@ -63,14 +69,26 @@ class ActivityTest {
     }
 
     @Test
-    fun `check setAsImmersive sets the correct flags`() {
+    fun `check setAsImmersive sets the correct flags  on devices with Android Q and below`() {
         activity.setAsImmersive()
 
         verify(decorView).systemUiVisibility = View.SYSTEM_UI_FLAG_FULLSCREEN
         verify(decorView).systemUiVisibility = View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
         verify(window.decorView, never()).setOnSystemUiVisibilityChangeListener(any())
     }
 
+    @Test
+    @Config(sdk = [Build.VERSION_CODES.R])
+    fun `check setAsImmersive sets the correct flags  on devices with Android R and above`() {
+        `when`(window.getInsetsController()).thenReturn(windowInsetsController)
+
+        activity.setAsImmersive()
+
+        verify(window).setDecorFitsSystemWindows(false)
+        verify(windowInsetsController).hide(WindowInsetsCompat.Type.systemBars())
+        verify(windowInsetsController).systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
```

Comment:
nit: my formatter is putting `WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE` on a newline